dle/data/util.py

```diff
@@ -83,3 +87,24 @@ def reformat_html_tags_in_raw_text(text: str) -> str:
     text = text.replace("</linkhtml>", "</a>")
 
     return text
+
+
+def magnitude(vector):
+    """Compute the magnitude of a vector so we can normalize to unit length"""
+    # See: https://github.com/elastic/elasticsearch/blob/main/docs/reference/mapping/types/dense-vector.asciidoc
+    return math.sqrt(sum(pow(element, 2) for element in vector))
+
+
+def compute_section_embedding(text: str, model, word_count=256, normalize=True) -> list[float]:
+    n_segments = 1 + len(text.split()) // word_count
+    vecs = np.zeros((n_segments, 768))
```

Comment:
What's the meaning of `768`?

Reply:
This is straight from Caleb's code, but it's the number of dimensions (vector length) in the embedding. To do any vector math we need to provide Elastic with query vectors of the same size as the indexed vectors. For dot product (preferred comparison metric for longish documents), all the indexed vectors have to be normalized to the unit vector as well.